### ISSUE TITLE
2.0.4 is broken when using Symfony 2.7

### ISSUE THREAD

**Issue by fbourigault** [csarrazi/CsaGuzzleBundle#118]
When Symfony 2.7 is used, the PHP dump container is invalid. This happens when no handler waas provided to any client. Using Symfony 2.8.1, the container is valid (maybe 2.8.0 is working too, but I haven't tested it).

I originaly reported the issue in https://github.com/symfony/symfony/issues/17353.

Probably the minimum version of `symfony/dependency-injection` should be `^2.8|^3.0`

**Comment by csarrazi** [comment]
Can you show your configuration?

Normally, this should work correctly using Symfony 2.7 (which we have been using on quite a few projects).

**Comment by fbourigault** [comment]
I'm using `$container->preprendExtensionConfig`:
```php
+        $container->prependExtensionConfig('csa_guzzle', [
+            'clients' => [
+                'client1' => [
+                    'config' => [
+                        'base_uri' => 'base_uri',
+                        'headers' => [
+                            'Authorization' => 'Bearer token',
+                            'Content-Type' => 'content_type_client_1'
+                        ]
+                    ]
+                ],
+                'client2' => [
+                    'config' => [
+                        'base_uri' => 'base_uri',
+                        'headers' => [
+                            'Authorization: Bearer token',
+                            'Content-Type' => 'content_type_client2'
+                        ]
+                    ]
+                ]
+            ]
+        ]);
```

From `app/console debug:config CsaGuzzleBundle`, i get:
```yml
csa_guzzle:
    clients:
        client1: { config: { base_uri: 'base_uri', headers: { Authorization: 'Bearer token', Content-Type: content_type_client1 } }, class: GuzzleHttp\Client, lazy: false, middleware: {  }, alias: null }
        client2: { config: { base_uri: 'base_uri', headers: { 0: 'Authorization: Bearer token', Content-Type: content_type_client2 } }, class: GuzzleHttp\Client, lazy: false, middleware: {  }, alias: null }
    profiler:
        enabled: true
        max_body_size: 65536
    logger:
        enabled: true
        service: null
        format: '{hostname} {req_header_User-Agent} - [{date_common_log}] "{method} {target} HTTP/{version}" {code} {res_header_Content-Length}'
        level: debug
    cache:
        enabled: false
        adapter: null
    mock:
        enabled: false
        mode: replay
```

**Comment by csarrazi** [comment]
I'll fix during the week. I now have a better view of the problem. It seems to come from the DefinitionDecorator, which doesn't fetch the class from the parent service.

**Comment by xabbuh** [comment]
@csarrazi Please let me know if you think that we should fix this in the `DefinitionDecorator` in Symfony or if you see this as a bug in your bundle (see symfony/symfony#17353 for the related Symfony issue). I am not quite sure right now what is the actual issue, but I would be happy to help you with a PR for Symfony if the actual issue is in the Symfony code.

**Comment by csarrazi** [comment]
Hi @xabbuh!

I know there was an issue in symfony/symfony which was fixed for 2.8 / 3.0.

The issue is that the decorator service doesn't have its `class` set. This is then transformed to `NULL` at some point container. Then, `NULL` is stripped from its first and last characters, which is transformed to `UL`, which generates an invalid class name, which cannot be instanciated (`new \UL()`).

Unfortunately, I don't quite remember where the original issue was. But regardless, we need a fix for this case, as I'm not really fond of bumping the Symfony version to 2.8.*.

**Comment by xabbuh** [comment]
@csarrazi I absolutely agree with you. However, the issue being fixed in Symfony 2.8/3.0 was related to the shared option which was only introduced recently with Symfony 2.8 (thus this cannot be the issue on Symfony 2.7). But sadly I am not sure what would be the most simplified example code to reproduce this issue on Symfony 2.7.

**Comment by csarrazi** [comment]
I should be able to check this tomorrow.

**Comment by csarrazi** [comment]
@fbourigault Could you provide me a composer.json file with your issue? That would be great to find the actual problem! :)

**Comment by fbourigault** [comment]
My failing composer.json is quite complicated, but here comes instructions to reproduce the issue:

* Create a new Symfony project `symfony new csa-guzzle-bundle-118 2.7.9`
* Remove the `config.platform.php` entry from composer.json...
* Install CsaGuzzleBundle `composer require csa/guzzle-bundle`
* Enable csa/guzzle-bundle in AppKernel.php
* Write the following content in `src/AppBundle/DependencyInjection/AppExtension.php`:
```php
<?php
namespace AppBundle\DependencyInjection;

use Symfony\Component\DependencyInjection\ContainerBuilder;
use Symfony\Component\DependencyInjection\Extension\PrependExtensionInterface;
use Symfony\Component\HttpKernel\DependencyInjection\Extension;

class AppExtension extends Extension implements PrependExtensionInterface
{

    public function prepend(ContainerBuilder $container)
    {
        $container->prependExtensionConfig('csa_guzzle', [
            'profiler' => '%kernel.debug%',
            'logger' => true,
            'clients' => [
                'client1' => [
                    'config' => [
                        'base_uri' => 'base_uri',
                        'headers' => [
                            'Authorization' => 'Bearer token',
                            'Content-Type' => 'content_type_client_1'
                        ]
                    ]
                ],
                'client2' => [
                    'config' => [
                        'base_uri' => 'base_uri',
                        'headers' => [
                            'Authorization: Bearer token',
                            'Content-Type' => 'content_type_client2'
                        ]
                    ]
                ]
            ]
        ]);
    }

    public function load(array $configs, ContainerBuilder $container)
    {
    }
}
```
* Clear the cache `php app/console cache:clear`

When clearing the cache you should get this exception:
```
[Symfony\Component\DependencyInjection\Exception\RuntimeException]  
  Cannot dump definition because of invalid class name (NULL)
```

**Comment by csarrazi** [comment]
Thanks! I'll check during the day! :)

**Comment by csarrazi** [comment]
Fixed by #120
